pkg/agent/flexvolume/controller.go

```diff
@@ -141,6 +132,7 @@ func (c *FlexvolumeController) Attach(attachOpts AttachOptions, devicePath *stri
 					attachment.MountDir = attachOpts.MountDir
 					attachment.PodNamespace = attachOpts.PodNamespace
 					attachment.PodName = attachOpts.Pod
+					attachment.ClusterName = attachOpts.ClusterName
```

Comment:
since you are storing the clusterName, we should get rid of the `parseClusterName()`. We can now just get it from the CRD object

Reply:
i don't think we can get rid of `parseClusterName()` still, because we call `GetAttachInfoFromMountDir()` from the flex driver's `mount` command before the CRD is ever created: https://github.com/rook/rook/blob/master/cmd/rookflex/cmd/mount.go#L63

Reply:
We can update the `GetAttachInfoFromMountDir() ` to get the cluster name from the crd instead of this function.

Reply:
I'm not sure I follow.  During the flex driver's mount, when it calls `GetAttachInfoFromMountDir()` the CRD doesn't exist yet, right?

Reply:
Oh you are right.

Reply:
Oh you are right.

Reply:
Oh you are right.

Reply:
Oh you are right.